### ISSUE TITLE
exec format error.

### ISSUE THREAD

**Issue by gyaneshwar** [StateFarmIns/terraform-aws-default-log-retention#12]
Note: The verbiage expressed here are that of the author and are not representative of State Farm.

**Describe the bug**

```
Error: Lambda function (default-log-retention-global-log-retention-setter) returned error: ({"errorType":"Runtime.InvalidEntrypoint","errorMessage":"RequestId: 00000-000000-00000000 Error: fork/exec /var/task/bootstrap: exec format error"})
with module.working.module.default_log_retention.aws_lambda_invocation.run_on_existing_groups[0]
on .terraform/modules/working.default_log_retention/tf-global-retention-setter.tf line 42, in resource "aws_lambda_invocation" "run_on_existing_groups":
resource "aws_lambda_invocation" "run_on_existing_groups" {
```

**Comment by Clete2** [comment]
Looking into it.

**Comment by Clete2** [comment]
@gyaneshwar sorry for the invoncenience. It appears that the GitHub Action created an invalid binary. I have manually created v1.0.3 which, in my local testing, fixes the issue.

What I did was build from my workstation, commit, tag, and push it up manually.

I will need to spend more time investigating why GitHub Actions built a bad binary. Could be an issue with the toolchain that it installs. Curiously 1.0.1 did not have this issue, and I didn't change anything in the interim with the build system.

Can you try v1.0.3 and let me know if it works?

By the way - I am leaving work early today but I will check back in later today/tomorrow.

**Comment by Clete2** [comment]
I created #13 to track the work for fixing the binary issue

**Comment by Clete2** [comment]
Hi @gyaneshwar. Checking in. Is your issue resolved?

**Comment by gyaneshwar** [comment]
> Hi @gyaneshwar. Checking in. Is your issue resolved?

it didn't work :(

**Comment by Clete2** [comment]
I'm sorry to hear that @gyaneshwar . Can you send me an updated error message or any other details that could help me troubleshoot?

**Comment by Clete2** [comment]
Hi @gyaneshwar, I released v1.0.5 today which permanently fixes this issue. The problem was I was using Amazon Linux 2 runtime but it had an old version of glibc, which was causing the error.

I just updated to use Amazon Linux 2023 which contains a proper version of glibc.

**Comment by Clete2** [comment]
I believe the issue is fixed, if it is not, please reopen and provide details.
